### PR TITLE
WW-434 | tracking discussions link within the 'On The Wiki' menu

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/WW-434

//cc @Wikia/west-wing

### CODE REVIEW

File: skins/oasis/js/Tracking.js [hunk 1]
```diff
@@ -671,25 +671,27 @@ jQuery(function ($) {
 			canonical = el.data('canonical');
 			if (canonical !== undefined) {
 				switch (canonical) {
```
Comment (WikiaTech) [inline skins/oasis/js/Tracking.js:673]: MAJOR Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* MAJOR 8 major
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/9aa8dadb34b384f2efb6956e3911d76b4c7cf80b/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/9aa8dadb34b384f2efb6956e3911d76b4c7cf80b/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/9aa8dadb34b384f2efb6956e3911d76b4c7cf80b/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L596](https://github.com/Wikia/app/blob/9aa8dadb34b384f2efb6956e3911d76b4c7cf80b/skins/oasis/js/Tracking.js#L596): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L616](https://github.com/Wikia/app/blob/9aa8dadb34b384f2efb6956e3911d76b4c7cf80b/skins/oasis/js/Tracking.js#L616): Function has a complexity of 12 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L658](https://github.com/Wikia/app/blob/9aa8dadb34b384f2efb6956e3911d76b4c7cf80b/skins/oasis/js/Tracking.js#L658): Function has a complexity of 17 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MINOR [Tracking.js#L596](https://github.com/Wikia/app/blob/9aa8dadb34b384f2efb6956e3911d76b4c7cf80b/skins/oasis/js/Tracking.js#L596): Replace this "switch" statement with "if" statements to increase readability. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1301)
1. INFO [Tracking.js#L262](https://github.com/Wikia/app/blob/9aa8dadb34b384f2efb6956e3911d76b4c7cf80b/skins/oasis/js/Tracking.js#L262): Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)
1. INFO [Tracking.js#L270](https://github.com/Wikia/app/blob/9aa8dadb34b384f2efb6956e3911d76b4c7cf80b/skins/oasis/js/Tracking.js#L270): Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)

File: skins/oasis/js/Tracking.js [hunk 2]
```diff
@@ -671,25 +671,27 @@ jQuery(function ($) {
...
+						label = 'on-the-wiki-chat';
+						break;
+					case 'forum':
+						label = 'on-the-wiki-forum';
+						break;
+					case 'videos':
+						label = 'on-the-wiki-videos';
+						break;
 				}
+			} else if (el.attr('href').indexOf('/d/f') != -1) {
```
Comment (idradm) [inline skins/oasis/js/Tracking.js]: 👍

**Comment by WikiaTech** [comment]
SonarQube analysis reported 10 issues
- MAJOR 7 major
- MINOR 1 minor
- INFO 2 info

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/a5a390be259b6cda91c306dde15c51fea5ed4995/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
2. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/a5a390be259b6cda91c306dde15c51fea5ed4995/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
3. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/a5a390be259b6cda91c306dde15c51fea5ed4995/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
4. MAJOR [Tracking.js#L596](https://github.com/Wikia/app/blob/a5a390be259b6cda91c306dde15c51fea5ed4995/skins/oasis/js/Tracking.js#L596): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
5. MAJOR [Tracking.js#L616](https://github.com/Wikia/app/blob/a5a390be259b6cda91c306dde15c51fea5ed4995/skins/oasis/js/Tracking.js#L616): Function has a complexity of 12 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
6. MAJOR [Tracking.js#L658](https://github.com/Wikia/app/blob/a5a390be259b6cda91c306dde15c51fea5ed4995/skins/oasis/js/Tracking.js#L658): Function has a complexity of 17 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
7. MINOR [Tracking.js#L596](https://github.com/Wikia/app/blob/a5a390be259b6cda91c306dde15c51fea5ed4995/skins/oasis/js/Tracking.js#L596): Replace this "switch" statement with "if" statements to increase readability. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1301)
8. INFO [Tracking.js#L262](https://github.com/Wikia/app/blob/a5a390be259b6cda91c306dde15c51fea5ed4995/skins/oasis/js/Tracking.js#L262): Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)
9. INFO [Tracking.js#L270](https://github.com/Wikia/app/blob/a5a390be259b6cda91c306dde15c51fea5ed4995/skins/oasis/js/Tracking.js#L270): Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)

**Review by idradm — APPROVED** [review]

**Review by lukaszkonieczny — APPROVED** [review]

**Pull request merged**